xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/displayers/defaultDisplayer.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+
+define([
+  "jquery",
+], function ($) {
+
+  /**
+   * Load the displayer custom css to the page
+   */
+  (function loadCss(url) {
+    var link = document.createElement("link");
+    link.type = "text/css";
+    link.rel = "stylesheet";
+    link.href = url;
+    document.getElementsByTagName("head")[0].appendChild(link);
+  })(BASE_PATH + "displayers/displayers.css");
+
+
+  /**
+   * Create a default displayer for a property of an entry
+   * @param {String} propertyId The property id of the entry to display
+   * @param {Object} entryData The entry to display the property of
+   * @param {Object} logic The logic object associated to the livedata
+   */
+  var Displayer = function (propertyId, entryData, logic) {
+
+    this.logic = logic;
+    this.propertyId = propertyId;
+    this.entryData = entryData;
+
+    this.element = document.createElement("div");
+    this.element.className = "livedata-displayer";
+
+    this.view();
```

Comment:
If you create the view in the constructor, how can I execute some code after the view is ready?

Reply:
How I see the displayer is like a small "plugin": the layout implement it and then it is almost autonomous. If the layout still wants to do things manually for some reasons, it can call the methods and then execute some code once its finished. I can also trigger events if you want that to be accessible from anywhere.

Reply:
We're talking about an API here, and an API that is async but doesn't allow you to execute code after, clearly has a problem. You make the assumption that the API will be used in a single way.